Source/Lib/Matroska/Matroska_Common.cpp

```diff
@@ -800,16 +812,61 @@ void matroska::Shutdown()
             TrackInfo_Current->FrameWriter.Mode[frame_writer::IsNotEnd] = false;
             TrackInfo_Current->FrameWriter.FrameCall(TrackInfo_Current->Frame.RawFrame, OutputFileName);
         }
+
+        // Checks
+        if (Errors)
+        {
+            if (!TrackInfo_Current->Unique && TrackInfo_Current->DPX_Buffer_Pos > TrackInfo_Current->DPX_Buffer_Count)
+            {
+                string OutputInfo = "Track " + to_string(i) + ", " + to_string(TrackInfo_Current->DPX_Buffer_Pos - TrackInfo_Current->DPX_Buffer_Count) + " frames";
+                if (TrackInfo_Current->DPX_Buffer_Count)
+                {
+                    string OutputFileName = string((const char*)TrackInfo_Current->DPX_FileName[TrackInfo_Current->DPX_Buffer_Count - 1], TrackInfo_Current->DPX_FileName_Size[TrackInfo_Current->DPX_Buffer_Count - 1]);
+                    FormatPath(OutputFileName);
+                    OutputInfo += " after ";
+                    OutputInfo += OutputFileName;
+                }
+                Errors->Error(IO_FileChecker, error::Undecodable, (error::generic::code)filechecker_issue::undecodable::Frame_Compressed_Extra, OutputInfo);
+            }
+            if (TrackInfo_Current->DPX_Buffer_Pos < TrackInfo_Current->DPX_Buffer_Count)
+            {
+                string OutputFileName = string((const char*)TrackInfo_Current->DPX_FileName[TrackInfo_Current->DPX_Buffer_Pos], TrackInfo_Current->DPX_FileName_Size[TrackInfo_Current->DPX_Buffer_Pos]);
+                FormatPath(OutputFileName);
+                Errors->Error(IO_FileChecker, error::Undecodable, (error::generic::code)filechecker_issue::undecodable::Frame_Compressed_Missing, OutputFileName);
+                 if (TrackInfo_Current->DPX_Buffer_Count - TrackInfo_Current->DPX_Buffer_Pos > 1)
+                {

```

Comment:
nit: alignment

Reply:
Fixed.
Note that you may want to do a new review, I added lot of code for handling old packages.

Reply:
I will try to test during the weekend on old packages.